nevergrad/functions/PBOfunctions/core.py

```diff
@@ -0,0 +1,108 @@
+# from IOHexperimenter import IOH_function 
+#TODO: figure out which way is better. importing the pbo_dict ignores all wrapper-functions, but it seems
+#like they are not needed here anyway, only the evaluate is actually used, so this should be faster
+#But, is is quite an ugly solution and should probably be improved
+from IOHexperimenter.src.IOH_function import pbo_fid_dict 
+from IOHexperimenter import W_Model_OneMax, W_Model_LeadingOnes
+import numpy as np
+import nevergrad as ng
+from nevergrad.common.typetools import ArrayLike
+from .. import base
+
+class PBO_function(base.ExperimentFunction):
+    """
+    Pseudo-boolean functions taken from the IOHexperimenter project
+    
+    Parameters
+    ----------
+    fid: int
+        function number (for a list of functions, please see the documentation)
+    iid: int
+        the instance of the function, specifies the transformations in variable and objective space
+    dim: int
+        The dimensionality of the problem. For function 21 and 23 (N-queens), this needs to be a perfect square
+
+
+    Notes
+    -----
+    The IOHexperimenter is part of the IOHprofiler project (IOHprofiler.github.io)
+
+    References
+    -----
+    For a full description of the available problems and used transformation methods,
+    please see "Benchmarking discrete optimization heuristics with IOHprofiler" by C. Doerr et al.
+    """
+
+    def __init__(self, fid: int = 1, iid: int = 0, dim: int = 16) -> None:
+        if fid in [21, 23]:
+            assert np.sqrt(dim).is_integer(), "Dimension needs to be a perfect square for the selected problem"
+#         self.f_internal = IOH_function(fid = fid, dim = dim, iid = iid, suite = "PBO")
+        self.f_internal = pbo_fid_dict[fid](iid, dim)
+        parameterization = ng.p.Array(shape = (dim, ) )
+        parameterization.set_integer_casting()
+        parameterization.set_bounds(0,1) #Not sure this is actually doing what I'm intending to achieve (0-1 inputs only)
```

Comment:
I would advise going for:
`ng.p.Choice([0, 1], repetitions=dim)`
This way you'll have only 0 and 1, and the optimizers will work on probabilities (so the same parametrization may actually need to different values)
you could also choose `ng.p.TransitionChoice([0, 1], repetitions=dim)`. With this one the optimizer will handle a value, which will lead to a 0 if it is negative, and 1 if it is positive. This way it is no more stochastic, but the problem becomes very non-continuous

Reply:
Thank you for this advice. 
I am just a little confused about how this parameterization is supposed to work. When I use this first option you recommend, it seems like the dimension becomes twice as large, leading to different expected inputs to the function (which expects to get a vector of {0,1}^dim). 
However, if I look at some of the existing pseud-boolean functions (I see that onemax and leadingones are present), they seem to work with either of these options (if I'm reading that correctly in the experiments.py file).
I think there is something I'm misunderstanding about the exact working of this parameterization object, but I can't quite figure out what.

Reply:
onemax and leadingones are somehow special cases which are processed differently because of historical reasons, they were here much before the parametrization system.
Indeed with `Choice` the size may be twice as big, because each case (`True` or `False`) gets a softmax weight, even though that's redundant. @teytaud should the case of 2 options be a special case (and I can update the code accordingly) or is there a more general method to avoid this redundancy?